### ISSUE TITLE
User reported issue: can't test or save translations, translations aren't published

### ISSUE THREAD

**Issue by oliver-phet** [phetsims/rosetta#262]
Sims: Build a Molecule, Collision Lab, Energy Skate Park, and Number Line: Operations

> I already translated those 4 sims twice and submitted them.
> 
> My problem was after translation I couldn't save and test it by myself.

Apr 8, 2021, 4:09 AM:
> I translated "Build a Molecule AGAIN, and submitted it.
> 
> But still I couldn't save or test it.
> 
> You'd better check out that problem first.

@jbphet @liam-mulhall can you guys investigate?

**Comment by jbphet** [comment]
@oliver-phet - What was the locale or language that this user was attempting to translate?  Knowing this will help us search through the logs for what went wrong.

**Comment by oliver-phet** [comment]
Whoops, KO korean!

**Comment by liam-mulhall** [comment]
@oliver-phet, did the person say when they were trying to translate?

**Comment by oliver-phet** [comment]
First email received Apr 5, 2021, 6:19 PM
Second email Apr 8, 2021, 4:09 AM

I would assume it was near those timestamps, but if it is important to diagnose the issue, I could ask the translator if they know exactly when they were trying to translate.

**Comment by liam-mulhall** [comment]
I just logged into phethelp and got the email address. It's easier that way. I'm going to make some notes for myself. At this point I'm just looking at logs after 3/30.

Here's one error the translator experienced:
```
Mar 30 01:20:49 phet-server.int.colorado.edu rosetta[29680]: error - Unable to get website user data. Error: Hostname/IP doesn't match certificate's altnames: "IP: 128.138.129.16 is not in the cert's list: "
Mar 30 01:20:49 phet-server.int.colorado.edu rosetta[29680]: error - Unhandled rejection. Error Message: Cannot read property 'data' of undefined
Mar 30 01:20:49 phet-server.int.colorado.edu rosetta[29680]: error - Error Stack: TypeError: Cannot read property 'data' of undefined
Mar 30 01:20:49 phet-server.int.colorado.edu rosetta[29680]: at ensureValidSession (/data/share/phet/rosetta/js/ensureValidSession.js:137:39)
```

**Comment by liam-mulhall** [comment]
It looks like longTermStringStorage.saveTranslatedStrings is sometimes not being called when a string submission is made.

**Comment by liam-mulhall** [comment]
I should check to see when this issue started. It would be nice to know if any these instances where `saveTranslatedStrings` isn't called initially if it's *ever* called.

We should see something like this:
```
Post request received. URL: /translate/sim/friction/uz.
saveTranslatedStrings called, simOrLibName = friction, locale = lo
```

But occasionally, we're seeing:

```
Post request received. URL: /translate/sim/friction/uz.
queuing string submission for friction_uz
Initiating build request for sim: friction, locale: uz.
```

**Comment by liam-mulhall** [comment]
# Next Steps

- [ ] Make sure Ox-Dev is running the same version as phet-server.
- [ ] Test submitting a form with no changes.
- [ ] If the logs match the issue case, push the debug statement to master and pull master from phet-server.
- [ ] Ask the translator to retry submitting strings so that we can watch the logs with the debug statement. (We're looking for a situation where `stringSavePromises.length === 0`. This would indicate that there's something wrong with detecting changes.)

**Comment by jbphet** [comment]
I just took a quick look at the logs to make sure other translations are being successfully submitted and published, and it seems like things are working fine.  I'm seeing an Armenia translation of Arithmetic on 4/14/2021 at 11:52 AM our time, an Uzbek translation of States of Matter on 4/14/2021 at 8:50 AM, and a German translation of Ratio and Proportion on 4/13/2021 at 11:01 PM.  I verified that each of these were published by going to the main website, loading the translated version, and checking the "About" dialog.

Bottom line: There doesn't appear to be anything fundamentally wrong with rosetta that is blocking all translations, so we need to figure out what is going on for this particular user.

**Comment by liam-mulhall** [comment]
I need to create a new branch of Babel.

**Comment by liam-mulhall** [comment]
Submitting a form with changes:
```
debug - Post request received. URL: /translate/sim/chains/ab.
info - queuing string submission for chains_ab
info - replacing line feed sequence in string from repo chains with key multilineString
debug - saveTranslatedStrings called, simOrLibName = chains, locale = ab
debug - Queued 1 promises for long-term string storage.
info - successfully committed changes for file chains/chains-strings_ab.json
info - build requests for translations are disabled, skipping
```

Submitting a form with no changes:
```
debug - Post request received. URL: /translate/sim/chains/ab.
info - queuing string submission for chains_ab
info - replacing line feed sequence in string from repo chains with key multilineString
debug - Queued 0 promises for long-term string storage.
info - build requests for translations are disabled, skipping
```

**Comment by liam-mulhall** [comment]
Email was sent to user asking them to submit a translation to do the test outlined in https://github.com/phetsims/rosetta/issues/262#issuecomment-816278743.

**Comment by liam-mulhall** [comment]
Both of the tests the user did have the following line.
```
debug - Queued 0 promises for long-term string storage.
```

**Comment by liam-mulhall** [comment]
I'm going to try to reproduce this issue on Ox-Dev with the tests branch of Babel. I will submit a BAM translation.

**Comment by liam-mulhall** [comment]
The user provided OS + browser versions:

> 1. What web browser are you using?
> * Chrome
> 2. What is the version of the web browser?
> * 90.0.4430.212 (64bit) - it's the latest version
> 3. What operating system are you using? 
> * Windows 10
> 4. What is the version of the operating system?
> * 2004

**Comment by liam-mulhall** [comment]
Further testing and log checking seems to be in order. I will coordinate with @KatieWoe since I don't have access to a Windows 10 device at the moment.

**Comment by jbphet** [comment]
I'm going to unassign myself.  @liam-mulhall and I have been checking in on this regularly, and it seems like there must have been something odd about the interaction that this particular user had.  We should have him re-submit and, once the problematic translations have been successfully submitted and published, we should be able to close this.

**Comment by oliver-phet** [comment]
@KatieWoe have you tested or identified any issues on Win10?

**Comment by KatieWoe** [comment]
@liam-mulhall correct me if I'm wrong, but I believe that the testing was done in https://github.com/phetsims/qa/issues/658 and it doesn't look like was anything was found there. Let me know what else we can do to help. @stemilymill can you confirm this was done on Win 10?

**Comment by stemilymill** [comment]
@KatieWoe yes, I did the testing in https://github.com/phetsims/qa/issues/658 on win10

**Comment by KatieWoe** [comment]
In that case, @liam-mulhall let us know what more we can do to help track this down. Would it help to set up a pair time to work in tandem?

**Comment by liam-mulhall** [comment]
> I believe that the testing was done in https://github.com/phetsims/qa/issues/658 and it doesn't look like was anything was found there

Yeah, that's the status of this issue. We don't know why it's happening and we haven't been able to reproduce the issue. Almost all of my PhET Rosetta time is being spent on the refactor.

> Would it help to set up a pair time to work in tandem?

Perhaps. @jbphet, do you think it's worth it to spend more time on this issue given that it could be resolved by the refactor?

**Comment by liam-mulhall** [comment]
Here are a couple of WAG's:

* bad character set?
* browser extensions messing with the form?
* browser settings?

**Comment by liam-mulhall** [comment]
More spitballing:

```
debug - Queued 0 promises for long-term string storage.
```

leads me to think there's something going on where the user is typing stuff in, but for whatever reason, when the form is submitted, it's somehow empty. What could do something like that? Is it possible that they have JavaScript disabled?

**Comment by liam-mulhall** [comment]
I think we should try doing a test where we disable JavaScript in Chrome and see if we can reproduce.

**Comment by oliver-phet** [comment]
@liam-mulhall I'm going to ask the translator to try disabling all extensions in an incognito window.

**Comment by jbphet** [comment]
@oliver-phet - Do we only have one translator for this locale?  If there is another, it would be good to reach out to them and have them try to submit.  It would help us to figure out whether the problem is unique to the one translator (which seems to be our leading theory at the moment), or something to do with the locale.

**Comment by oliver-phet** [comment]
Only one translator. But couldn't any of us test submitting a Korean translation?

**Comment by liam-mulhall** [comment]
> couldn't any of us test submitting a Korean translation?

We did and we can't reproduce the issue.

**Comment by liam-mulhall** [comment]
@oliver-phet, you should also suggest trying a different browser and (if possible) a different computer.

**Comment by oliver-phet** [comment]
@liam-mulhall ko translator reported back that Edge was tried and the same error (submitted, but the babel file doesn't exist; can't test or save).

I'm suspecting Javascript might be disabled (in both browsers?) I'm going to request console output on test/save to see if that offers any clues.

**Comment by liam-mulhall** [comment]
To disable JavaScript, you have to go to settings and toggle it off. I'm pretty sure it's enabled by default on Chrome and Edge.

It would be weird if the user disabled it because it would break a lot of websites.

**Comment by oliver-phet** [comment]
I just re-pinged the translator. Our newest sims still haven't been translated into _ko so I suspect the issue is still blocking translation.

**Comment by oliver-phet** [comment]
Latest from _ko translator:

> I have tried to translate the SIMS that are not untranslated into Korean..
> 
> But still I can not "SAVE" or "TEST" the translation. 
> 
> Though I can "SUBMIT" it but it won't  be reflected.

**Comment by liam-mulhall** [comment]
I will discuss this with JB, and we'll try to figure out what to do.

**Comment by liam-mulhall** [comment]
Neither JB nor I really know where to go with this issue. We didn't have enough time to discuss in-depth, so I add this to the agenda for our next meeting.

**Comment by liammulh** [comment]
There is nothing to do except monitor for this in Rosetta 2.0. I am going to un-assign current assignees.
